tsl/src/reorder.c

```diff
@@ -148,12 +150,45 @@ tsl_move_chunk(PG_FUNCTION_ARGS)
 				 errmsg("valid chunk, destination_tablespace, and index_destination_tablespaces "
 						"are required")));
 
-	reorder_chunk(chunk_id,
-				  index_id,
-				  verbose,
-				  wait_id,
-				  destination_tablespace,
-				  index_destination_tablespace);
+	chunk = ts_chunk_get_by_relid(chunk_id, false);
+
+	if (NULL == chunk)
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("\"%s\" is not a chunk", get_rel_name(chunk_id))));
+
+	if (ts_chunk_contains_compressed_data(chunk, &chunk_parent))
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("cannot directly move internal compression data"),
+				 errdetail("Chunk \"%s\" contains compressed data for chunk \"%s\" and cannot be "
+						   "moved directly.",
+						   get_rel_name(chunk_id),
+						   get_rel_name(chunk_parent->table_id)),
+				 errhint("Moving chunk \"%s\" will also move the compressed data.",
+						 get_rel_name(chunk_parent->table_id))));
+
+	/* If chunk is compressed move it by altering tablespace on both chunks */
+	if (OidIsValid(chunk->fd.compressed_chunk_id))
+	{
+		Chunk *compressed_chunk = ts_chunk_get_by_id(chunk->fd.compressed_chunk_id, true);
+		AlterTableCmd cmd = { .type = T_AlterTableCmd,
+							  .subtype = AT_SetTableSpace,
+							  .name = get_tablespace_name(destination_tablespace) };
+
+		AlterTableInternal(chunk_id, list_make1(&cmd), false);
+		AlterTableInternal(compressed_chunk->table_id, list_make1(&cmd), false);
```

Comment:
If the chunk data is compressed and a valid index  is passed, I think we should print a NOTICE saying that the index is ignored as the data is compressed.